addon/-private/system/relationships/relationship-payloads.js

```diff
@@ -324,7 +324,7 @@ export default class RelationshipPayloads {
       return;
     }
 
-    if (Array.isArray(data)) {
+    if (data.length) { // dirty-check for "is-array"
```

Comment:
I'd recommend to stick to `Array.isArray`, which gives you predictable performance, unless you now for sure that `data` is always an array.